kubernetes/structures_tokenrequest.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package kubernetes
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	v1 "k8s.io/api/authentication/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Flatteners
+
+func flattenTokenRequestSpec(in v1.TokenRequestSpec, d *schema.ResourceData, meta interface{}) ([]interface{}, error) {
+	att := make(map[string]interface{})
+
+	att["audiences"] = in.Audiences
+
+	if in.BoundObjectRef != nil {
+		bndObjRef, err := flattenBoundObjectReference(*in.BoundObjectRef, d, meta)
+		if err != nil {
+			return nil, err
+		}
+		att["boundobjectref"] = bndObjRef
+	}
+
+	att["expirationseconds"] = int(*in.ExpirationSeconds)
```

Comment:
`expiration_seconds`

You also want to check that `in.ExpirationSeconds` is not null before dereferencing it. It works here because the API sends back a default value for this field, but if it were to come back empty then this code would panic.